server/src/main/java/org/elasticsearch/index/seqno/LocalCheckpointTracker.java

```diff
@@ -158,6 +158,21 @@ public synchronized void waitForOpsToComplete(final long seqNo) throws Interrupt
         }
     }
 
+    /**
+     * Checks if the given sequence number was marked as completed in this tracker.
+     */
+    public synchronized boolean contains(long seqNo) {
```

Comment:
I don't think we need to be fully synchronized here (and so can save ourselves from locking for the quick returns). I think that we only need to lock on this on access to `processedSeqNo`. I see that we have asserts that we have a lock on this in `getBitSetKey` and `seqNoToBitSetOffset` but I think that neither of those are necessary.

Reply:
Good suggestion. I pushed https://github.com/elastic/elasticsearch/pull/33871/commits/d22f136a757e47fd427a5a897c9603313cfc5d5d.